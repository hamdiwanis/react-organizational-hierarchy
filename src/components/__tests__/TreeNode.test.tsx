import React from 'react';
import { render } from '@testing-library/react';
import serializer from 'jest-emotion';
import { TreeNode } from '../../index';

expect.addSnapshotSerializer(serializer);

describe('TreeNode', () => {
  it('is exported', () => {
    expect(TreeNode).toBeTruthy();
  });

  it('Renders', () => {
    const component = render(<TreeNode label={'test'} />);
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
