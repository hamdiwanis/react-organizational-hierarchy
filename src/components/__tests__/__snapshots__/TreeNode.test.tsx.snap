// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeNode Renders 1`] = `
.emotion-0 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  text-align: center;
  list-style-type: none;
  position: relative;
  padding: var(--tree-line-height) var(--tree-node-padding) 0 var(--tree-node-padding);
}

.emotion-0::before,
.emotion-0::after {
  content: '';
  position: absolute;
  top: 0;
  height: var(--tree-line-height);
  box-sizing: border-box;
  right: 50%;
  width: 50%;
  border-top: var(--tree-line-width) var(--tree-node-line-style) var(--tree-line-color);
}

.emotion-0::after {
  left: 50%;
  border-left: var(--tree-line-width) var(--tree-node-line-style) var(--tree-line-color);
}

.emotion-0:only-of-type {
  padding: 0;
}

.emotion-0:only-of-type::after,
.emotion-0:only-of-type:before {
  display: none;
}



.emotion-0:first-of-type::before {
  border: 0 none;
}

.emotion-0:first-of-type::after {
  border-radius: var(--tree-line-border-radius) 0 0 0;
}



.emotion-0:last-of-type::before {
  border-right: var(--tree-line-width) var(--tree-node-line-style) var(--tree-line-color);
  border-radius: 0 var(--tree-line-border-radius) 0 0;
}

.emotion-0:last-of-type::after {
  border: 0 none;
}

<li
  class="emotion-0"
>
  test
</li>
`;
